hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AExceptionTranslation.java

```diff
@@ -114,8 +116,11 @@ public void test404isNotFound() throws Exception {
    */
   @Test
   public void testUnknownBucketException() throws Exception {
-    AmazonS3Exception ex404 = createS3Exception(SC_404);
-    ex404.setErrorCode(ErrorTranslation.AwsErrorCodes.E_NO_SUCH_BUCKET);
+    S3Exception ex404 = createS3Exception(b -> b
+        .statusCode(SC_404)
+        .awsErrorDetails(AwsErrorDetails.builder()
+            .errorCode(ErrorTranslation.AwsErrorCodes.E_NO_SUCH_BUCKET)
```

Comment:
I think when I was working on this, I did test that the NoSuchBucket Error code is still thrown, but I am wondering if you tested too? Otherwise can add to todo to verify

Reply:
Yes, still there. Also tested in testNoBucketProbing()